### ISSUE TITLE
Concurrency Issue

### ISSUE THREAD

**Issue by jlareo** [doghappy/socket.io-client-csharp#176]
By removing the lock (https://github.com/doghappy/socket.io-client-csharp/commit/6d1c5d9cf6a078644f2b7df9e966b196ce1355c5), could not be a problem in the future if two threads try to perform an emit at the same time (by having a change of context) ?

**Comment by doghappy** [comment]
`System.Net.Websockets.ClientWebSocket` does not allow sending multiple messages at the same time.

When `SocketIO` emit data, it may also send a ping to the socket.io server v2.x at the same time. An exception will be thrown at this time.

So, I added the lock to the lower-level IWebSocketClient. Removed the lock in SocketIO.

The SocketIO class is not thread-safe, the instance methods share these data:

![image](https://user-images.githubusercontent.com/20038914/126122439-f63f397a-9ba9-4f64-a326-d569ba61f0ea.png)


When users try to emit data in multiple threads at the same time, there will be problems, none of them are thread-safe.
